src/extension/provider/runmeTask.ts

```diff
@@ -63,36 +86,51 @@ export class RunmeTaskProvider implements TaskProvider {
     return task
   }
 
-  static getRunmeTask (filePath: string, command: string, options: TaskOptions = {}): RunmeTask {
+  static async getRunmeTask (
+    filePath: string,
+    command: string,
+    cell: NotebookCell|NotebookCellData,
+    options: TaskOptions = {},
+    runner: IRunner,
+    environment?: IRunnerEnvironment,
+  ): Promise<Task> {
+    const source = workspace.workspaceFolders?.[0] ?
+      path.relative(workspace.workspaceFolders[0].uri.fsPath, filePath) :
+      path.basename(filePath)
+
+    const { interactive, background } = getAnnotations(cell.metadata)
+
     const cwd = options.cwd || path.dirname(filePath)
-    const closeTerminalOnSuccess = options.closeTerminalOnSuccess || true
-    const isBackground = options.isBackground || false
-
-    const definition: RunmeTaskDefinition = {
-      type: 'runme',
-      filePath,
-      command,
-      closeTerminalOnSuccess,
-      isBackground,
-      cwd
+    const isBackground = options.isBackground || background
+
+    const runOpts: RunProgramOptions = {
```

Comment:
Do we have to "seed" the `envs` here with the host processes? Or does the runner take of that?

Reply:
Oh yeah, we have to seed it, good call.

In the future I'd like to integrate this with the existing session.

Reply:
Makes sense.

Reply:
Added support for environments and env population in 452eb96 @sourishkrout